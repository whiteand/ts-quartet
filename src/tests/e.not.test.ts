import { e as v, ExplanationSchemaType } from '..'
import { getExplanations } from './getExplanations'
import { testValidatorImpure } from './testValidatorImpure'

describe('v.not', () => {
  test('v.not', () => {
    testValidatorImpure(
      v(v.not(false)),
      [null, [], {}, 1, 0, NaN, undefined, true],
      [false],
    )
    expect(getExplanations(v(v.not(false)), false)).toEqual([
      {
        path: [],
        schema: {
          schema: {
            type: ExplanationSchemaType.Primitive,
            value: false,
          },
          type: ExplanationSchemaType.Not,
        },
        value: false,
        innerExplanations: [],
      },
    ])
  })
})
