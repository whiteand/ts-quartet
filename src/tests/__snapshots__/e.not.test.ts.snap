// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`v.not v.not 1`] = `
Array [
  Object {
    "path": Array [],
    "schema": Object {
      "schema": Object {
        "type": "Primitive",
        "value": false,
      },
      "type": "Not",
    },
    "value": false,
  },
]
`;
